### DISCUSSION TITLE
Can't Flash/Erase Flash on Esp32

### DISCUSSION THREAD

**Discussion by VikiAgung** [micropython/micropython#12902] [ESP32] [answered]
Hi everyone, I'm new with this and i have an esp32 that won't flash via command prompt. Even the Boot button doesn't work during the flash process. This is the output i got:
C:\Users\user>esptool --chip esp32 -p COM4 erase_flash
esptool.py v4.6.2
Serial port COM4
Connecting......................................

A fatal error occurred: Failed to connect to ESP32: No serial data received.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html

Does anyone know the problem? And what should i do?

**Comment by karfas** [comment]
This is not a micropython issue. 

Which kind of device do you use ?
What have you already tried from the steps in your troubleshooting link ?

Most likely, the ESP32 doesn't enter download mode. 
All dev boards I have seen provide a button which might need to pressed once (or during reset). 

It's also possible that the USB cable you are using is a "bad" one. Try another.

**Reply by VikiAgung** [reply]
My device is ESP32 DEVKITV1. I tried several steps like as updating the CP210 USB to UART driver and pressing the Boot and EN button while flashing. The USB cable worked fine when I tried it with the ESP8266, erasing flash or flashing the micropython firmware on the ESP8266. And I have tried using another USB cable and the results are still the same.

**Comment by robert-hh** [comment, marked as answer]
You should as well confirm that COM4 is the right port. You can test that in the device manager. The port must appear when you plug in the ESP32 board, and disappear when you unplug it. If there is no change at all, the cable might be wrong.
Since the boot switch just connects GPIO0 to GND, you can get the same result by connecting the GPIO0 pin of the board to GND and push reset.

**Reply by VikiAgung** [reply]
I have checked and confirmed the port correctly via Device Manager or Command Promp using "mode" command and the output is: 

Status for device COM4: 
----------------------- 
Baud: 115200 
Parity: None 
Data Bits: 8 
Stop Bits: 1 
Timeout: OFF 
XON/XOFF: OFF 
CTS handshaking: OFF 
DSR handshaking: OFF 
DSR sensitivity: OFF 
DTR circuit: OFF 
RTS circuit: OFF 

I have also tried using different usb cables and the results are still same. Maybe, I will try to connecting GPIO0 to GND. Thanks for ur answer :)

**Reply by VikiAgung** [reply]
The problem was solved. I used the Expansion Base for the esp32 then connected the GND/5V pin to the EN and D2 pins. After successfully with this step, I was immediately able to detect Micropython on the ESP32 using Thonny without needing to erase it or re-flash the MicroPython firmware. Thanks :)

**Comment by sophiedegran** [comment]
All is ok but i think you forget to put the board in boot mode ... you have to press a button before plug usb

**Reply by robert-hh** [reply]
You have to press & hold the boot button and then plug USB or push reset. Usually ESP boards have a hardware logic for this sequence which is used by esptool.py

**Comment by VikiAgung** [comment]
Thanks y'all for answering my problem :)
